torchvision/transforms/transforms.py

```diff
@@ -47,13 +51,30 @@ class Compose(object):
     """
 
     def __init__(self, transforms):
-        self.transforms = transforms
+        super().__init__()
 
-    def __call__(self, img):
+        try:
+            # Try to wrap the transforms that are not nn.Module into nn.Lambda
+            # register all transforms as nn.ModuleList <-> all transforms should be nn.Module
+            # if something fails we keep previous behaviour
+            new_transforms = []
+            for t in transforms:
+                if callable(t) and not isinstance(t, torch.nn.Module):
+                    t = Lambda(t)
+                new_transforms.append(t)
+
+            self.transforms = torch.nn.ModuleList(new_transforms)
```

Comment:
What is the motivation for using a `ModuleList` here? Given that `_forward_impl` simply passes the input sequentially through all transforms, would a `Sequential` a better fit?

Reply:
Here is another point for using `ModuleList` instead of `Sequential`. Previously, `Compose.transforms` is of the same type as input transforms: tuple, list. User can create transforms as a compose and then append/extend etc its content. For example, https://github.com/facebookresearch/swav/blob/master/src/multicropdataset.py#L47